lib/src/piece/sequence.dart

```diff
@@ -3,23 +3,41 @@
 // BSD-style license that can be found in the LICENSE file.
 
 import '../back_end/code_writer.dart';
+import '../constants.dart';
 import 'piece.dart';
 
 /// A piece for a series of statements or members inside a block or declaration
 /// body.
 ///
 /// Usually constructed using a [SequenceBuilder].
 class SequencePiece extends Piece {
+  /// The opening delimiter, if any.
+  final Piece? _leftBracket;
+
   /// The series of members or statements.
   final List<SequenceElement> _elements;
 
-  SequencePiece(this._elements);
+  SequencePiece(this._elements, {Piece? leftBracket, Piece? rightBracket})
+      : _leftBracket = leftBracket,
+        _rightBracket = rightBracket;
+
+  /// The closing delimiter, if any.
+  final Piece? _rightBracket;
 
-  /// Whether this sequence has any contents.
-  bool get isNotEmpty => _elements.isNotEmpty;
+  @override
+  List<State> get additionalStates => [if (_elements.isNotEmpty) State.split];
 
   @override
   void format(CodeWriter writer, State state) {
+    writer.setAllowNewlines(state == State.split);
+
+    if (_leftBracket case var leftBracket?) {
+      writer.format(leftBracket);
+      writer.splitIf(state == State.split,
+          space: false,
+          indent: _elements.isNotEmpty ? _elements.first.indent : 0);
```

Comment:
How about

```suggestion
          indent: _elements.firstOrNull?.indent ?? 0);
```

https://api.dart.dev/stable/3.2.6/dart-collection/IterableExtensions/firstOrNull.html

Reply:
Nice! Done.